cgc_bowtie2_build.py

```diff
@@ -0,0 +1,165 @@
+#!/usr/bin/env python
+
+#-----------------------------------------------------------------------------
+# Copyright (c) 2016--, Evguenia Kopylova, Jad Kanbar
+#
+# Distributed under the terms of the Modified BSD License.
+#
+# The full license is in the file COPYING.txt, distributed with this software.
+#-----------------------------------------------------------------------------
+
+"""
+Build Bowtie2 database on all reference genomes in Kraken report.
+"""
+
+import click
+import collections
+
+def load_kraken_mpa_report(kraken_mpa_report_fp,
+                           taxonomic_rank,
+                           taxa_levels,
+                           taxa_levels_idx,
+                           read_per_taxa):
+    """Absolute abundance of number of reads matching a defined taxa level.
+    Parameters
+    ----------
+    kraken_translate_report_fp: str
+        filepath to output of "kraken translate"
+    taxonomic_rank: str
+        taxonomy level (e.g., genus or species)
+    taxa_levels: dict
+        keys are full name taxonomic ranks and values are abbreviated ranks
+    taxa_levels_idx: dict
+        2-way dict storing integer depths for abbreviated taxonomic ranks
+    read_per_taxa: in
+        integer of number of minimum number of reads to keep per taxa
+
+    Returns
+    -------
+    taxonomies: set
+        set of taxonomies from kraken_translate_report_fp
+    split_on_level: str
+        string that determines the level to split the taxonomy to be used in
+        create_db_folder function
+    repo_file_name: str
+        string with filename suffix from kraken_translate_report_fp to be used
+        in create_db_folder function
+
+    """
+    repo_file_name = kraken_mpa_report_fp.strip().split('/')[-1].split('mpa')[0]
+
+    taxonomic_rank_level_str = taxa_levels[taxonomic_rank]
```

Comment:
Could this be moved to the ```main``` function and you pass only the strings ```taxonomic_rank_level_str``` and ```taxonomic_rank_level_int``` to ```load_kraken_mpa_report``` instead of the dicts ```taxa_levels``` and ```taxa_levels_idx```?